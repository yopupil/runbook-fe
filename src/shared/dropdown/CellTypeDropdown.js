import React from 'react';
import PropTypes from 'prop-types';
import Dropdown from 'shared/dropdown/Dropdown';
import CollapsedLabel from 'shared/labels/CollapsedLabel';

export default class CellDropdown extends Dropdown {
  render () {
    const { open } = this.state;
    const { actions } = this.props;
    return (
      <div className='cell-dropdown' ref={el => (this.container = el)}>
        <i
          className='zmdi zmdi-more-vert zmdi-hc-lg cell-dropdown__icon'
          onClick={this.toggleOpen}
        />
        {open ? (
          <div>
            <ul className='cell-dropdown__list' />
          </div>
        ) : (
          <React.Fragment>
            <CollapsedLabel isCircular={true} text={'js'} />
            <CollapsedLabel text={'py3nameforme'} />
          </React.Fragment>
        )}
      </div>
    );
  }

  toggleOpen () {
    this.setState({
      open: !this.state.open
    });
  }

  handleAction (action) {
    this.setState({
      open: false
    });
    this.props.handleAction(action);
  }
}

CellDropdown.propTypes = {
  actions: PropTypes.arrayOf(PropTypes.string),
  // The callback function to execute when action is triggered
  handleAction: PropTypes.func.isRequired
};

CellDropdown.defaultProps = {
  actions: []
};
