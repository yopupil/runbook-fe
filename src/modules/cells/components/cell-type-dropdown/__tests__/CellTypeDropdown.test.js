import React from 'react';
import { shallow } from 'enzyme';
import CellTypeDropdown from '../CellTypeDropdown.component';
import renderer from 'react-test-renderer';

function NOOP () {}

test('CellTypeDropdown.component snapshots', () => {
  let component = renderer.create(
    <CellTypeDropdown
      name='test'
      selections={['a', 'b']}
      selectedItem={'a'}
      onSelect={NOOP}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component = renderer.create(<CellTypeDropdown name='test' onSelect={NOOP} />);
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component = renderer.create(
    <CellTypeDropdown
      name='test'
      selections={['a', 'b']}
      selectedItem={'a'}
      labelAccessor={d => 'Label::' + d}
      onSelect={NOOP}
    />
  );
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('CellTypeDropdown.component.rendering', () => {
  let component = shallow(
    <CellTypeDropdown
      name='test'
      selections={['a', 'b']}
      selectedItem={'a'}
      labelAccessor={d => 'Label::' + d}
      onSelect={NOOP}
    />
  );

  const LIST_SELECTOR = 'ul.cell-type-dropdown__list';
  const LIST_ITEM_SELECTOR = 'li.cell-type-dropdown__list-item';
  const collapsedLabel = component.find('CollapsedLabel');

  expect(collapsedLabel.length).toEqual(1);
  expect(
    collapsedLabel.find({
      text: 'Label::a'
    }).length
  ).toEqual(1);

  // dropdpown is closed by default
  expect(component.find(LIST_SELECTOR).length).toEqual(0);

  collapsedLabel.simulate('click');

  const list = component.find(LIST_SELECTOR);

  expect(list.length).toEqual(1);

  // All items are shown
  const matches = component.find(LIST_ITEM_SELECTOR);
  expect(matches.length).toEqual(3);
  expect(matches.first().text()).toBe('test');

  expect(matches.at(1).text()).toBe('Label::a');
  expect(matches.at(2).text()).toBe('Label::b');

  // Trigger click inside
  list.simulate('click');

  expect(component.find(LIST_SELECTOR).length).toEqual(1);
});
