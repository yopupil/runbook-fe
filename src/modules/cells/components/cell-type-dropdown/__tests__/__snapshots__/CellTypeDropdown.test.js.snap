// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CellTypeDropdown.component snapshots 1`] = `
<div
  className="cell-type-dropdown"
>
  <label
    className="collapsed-label"
    onClick={[Function]}
    style={
      Object {
        "color": "#e57373",
      }
    }
  >
    a
  </label>
</div>
`;

exports[`CellTypeDropdown.component snapshots 2`] = `
<div
  className="cell-type-dropdown"
>
  <label
    className="collapsed-label"
    onClick={[Function]}
    style={
      Object {
        "color": "#e57373",
      }
    }
  >
    *
  </label>
</div>
`;

exports[`CellTypeDropdown.component snapshots 3`] = `
<div
  className="cell-type-dropdown"
>
  <label
    className="collapsed-label"
    onClick={[Function]}
    style={
      Object {
        "color": "#4db6ac",
      }
    }
  >
    Label::a
  </label>
</div>
`;
